src/utils/azure.ts

```diff
@@ -0,0 +1,39 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { BaseResource } from 'ms-rest-azure';
+import { localize } from '../localize';
+
+function parseResourceId(id: string): RegExpMatchArray {
+    const matches: RegExpMatchArray | null = id.match(/\/subscriptions\/(.*)\/resourceGroups\/(.*)\/providers\/(.*)\/(.*)/);
+
+    if (matches === null || matches.length < 3) {
+        throw new Error(localize('azFunc.InvalidResourceId', 'Invalid Azure Resource Id'));
+    }
+
+    return matches;
+}
+
+export function getResourceGroupFromId(id: string): string {
+    return parseResourceId(id)[2];
+}
+
+export function getSubscriptionFromId(id: string): string {
+    return parseResourceId(id)[1];
+}
+
+export function getNameFromId(id: string): string {
+    return parseResourceId(id)[4];
+}
+
+export interface IBaseResourceWithName extends BaseResource {
```

Comment:
To be clear, I'm okay with you copying over functions like `getSubscriptionFromId` because they're related to `getResourceGroupFromId` which you actually need, but `IBaseResourceWithName` and `IResourceResult` are only used by `promptForResource` from the functions repo and I don't think app service will ever need that